src/concurrency/thread.rs

```diff
@@ -188,22 +186,17 @@ impl<'mir, 'tcx> Thread<'mir, 'tcx> {
 #[derive(Debug)]
 pub enum Time {
     Monotonic(Instant),
-    Virtual {
-        /// The instant for this moment.
-        instant: u64,
-        /// A reference to the time anchor used to read the current global virtual time.
-        time_anchor: Arc<AtomicU64>,
-    },
+    Virtual { nanoseconds: u64 },
     RealTime(SystemTime),
 }
 
 impl Time {
     /// How long do we have to wait from now until the specified time?
-    fn get_wait_time(&self) -> Duration {
+    fn get_wait_time(&self, clock: &Clock) -> Duration {
         match self {
             Time::Monotonic(instant) => instant.saturating_duration_since(Instant::now()),
-            Time::Virtual { instant, time_anchor } =>
-                Duration::from_nanos(instant - time_anchor.load(Ordering::Relaxed)),
+            Time::Virtual { nanoseconds } =>
+                Duration::from_nanos(nanoseconds - clock.assert_virtual().load(Ordering::Relaxed)),
```

Comment:
Looks like `Clock` should have a `get_wait_time` method abstracting this -- then we don't need to expose the underlying clock value ever.